### PR TITLE
Added profiling duration to log formatted message

### PR DESCRIPTION
**Pull request by nikashitsa**
This Pull Request solves #9458.
P.S. I hope my Japanese translation is correct :)

### CODE REVIEW

**Comment by cebe** [comment]
it should not be logged by default. in most cases it is not relevant information.

**Comment by nikashitsa** [comment]
By default it will produce `[-]` for duration value. I think, log format should be the same for profiling messages and for other log messages. @samdark could you please bring your expertise here?

File: framework/log/Logger.php [hunk 1]
```diff
@@ -311,4 +332,22 @@ public static function getLevelName($level)
...
+
+    /**
+     * Calculates the elapsed time for the given profile end token.
+     * @param string $endToken token for the code block.
+     * @param float $time the profile end timestamp.
+     * @return float the elapsed time in seconds.
+     */
+    private function calculateProfileDuration($endToken, $time)
+    {
+        foreach ($this->_profileBeginnings as $i => $beginning) {
```
Comment (cebe) [inline framework/log/Logger.php]: as far as I see you could index the `$this->_profileBeginnings` array by `$token` and then select the necessary token to calculate timing instaead of iterating over the array. Or am I missing something?

**Comment by samdark** [comment]
Solved in 2.1 https://github.com/yiisoft/yii2/pull/14611

**Comment by samdark** [comment]
Won't fix in 2.0.

**Pull request closed without merging**